[fio toup] ostreemanager: Don't skip ostree pull on partial commits

detsch opened the pull request:
When running `ostree fsck --delete`, commits that have corrupted objects are be marked as partial. This change allows aktualizr to be used to recover from such situations.

detsch committed:
[fio toup] ostreemanager: Don't skip ostree pull on partial commits

When running `ostree fsck --delete`, commits that have corrupted objects
are be marked as partial. This change allows aktualizr to be used to
recover from such situations.

Signed-off-by: Andre Detsch <andre.detsch@foundries.io>

mike-sul commented on src/libaktualizr/package_manager/ostreemanager.cc:
```diff
@@ -93,13 +93,12 @@ data::InstallationResult OstreeManager::pull(const boost::filesystem::path &sysr
+  OstreeRepoCommitState commit_state;
+  if (ostree_repo_load_commit(repo.get(), refhash.c_str(), nullptr, &commit_state, &error)) {
+    if (commit_state & OSTREE_REPO_COMMIT_STATE_PARTIAL || commit_state & OSTREE_REPO_COMMIT_STATE_FSCK_PARTIAL) {
+      LOG_INFO << "refhash " << refhash << " is partially pulled";
```
Maybe add `... is partially pulled, re-pulling it...`?

mike-sul replied:
I wonder if we need to check whether an ostree commit/refhash is present at all and always re-pull as long as a client decides to call `OstreeManager::pull`, or make it configurable, so a client can always enforce re-pulling.

detsch replied:
Always re-pulling should be fine, it was my first approach to the problem. I don't see much of a reason to do the check, but I've tried minimize the behavior changes.
I also don't see a reason to make it configurable.

mike-sul replied:
ok, let's leave it as it is now.

mike-sul commented:
Maybe it makes sense to mimic/clone the same logic/change in the `OstreeManager::verifyTargetInternal` function here https://github.com/foundriesio/aktualizr/blob/a55d67eb44991fbb97ff1e58e3b5ebacad50653b/src/libaktualizr/package_manager/ostreemanager.cc#L351.

detsch commented:
> Maybe it makes sense to mimic/clone the same logic/change in the `OstreeManager::verifyTargetInternal` function here
> 
> https://github.com/foundriesio/aktualizr/blob/a55d67eb44991fbb97ff1e58e3b5ebacad50653b/src/libaktualizr/package_manager/ostreemanager.cc#L351

Yes, it does. Even if we remove the check inside pull, it makes sense to udpate the logic above. I'll do it.

detsch committed:
[fio toup] ostreemanager: Consider commit state when verifying target

If an ostree commit is only partially fetched, it should not be
considered good.

Signed-off-by: Andre Detsch <andre.detsch@foundries.io>
